### PR TITLE
Dispatched a custom event for chromosome selection from shadow DOM

### PR DESCRIPTION
**Pull request by YangQi007**

### CODE REVIEW

**Comment by anderspitman** [comment]
@YangQi007 this is looking great. As I played with it, I realized 3 things:

1. We also need to indicate when the user clicks "All". For this let's use `_all` for the name.
2. We should just implement generic region events now.
3. We should probably use "reference name" instead of chromosome, since that's what the file formats use.

It would look like this:

```json
{
  refName: <chromosome name>,
  start: <start>,
  end: <end>
}
```

If the user selected a whole chromosome, just leave start/end undefined.

We also want to emit this events when the user brushes, but that doesn't have to be done right now unless you want to.

**Comment by YangQi007** [comment]
What do you mean by "generic region events"? 
See the screenshot below after I changed the content of detail, please let me know if they're what you want. 
Screenshot 2024-06-27 at 4 18 11 PM

**Comment by anderspitman** [comment]
Generic region events means it covers all 3 cases:

1. user selects "All":

```javascript
{
  refName: "_all"
}
```

2. User selects a specific chromosome:

```javascript
{
  refName: "chr1"
}
```

3. User brushes any region, either from "All" or from a chromosome view:

```javascript
{
  refName: "chr1",
  start: 1234,
  end: 5678
}
```

Note that you shouldn't set start/end in the first two cases. Just set refName.

**Comment by YangQi007** [comment]
For the third case, in the "All" mode, when the user brushes the region, how will we handle the case of crossing several chromosomes? For example, the user may brush to cross several chromosomes.

**Comment by anderspitman** [comment]
That's a great point. Let's not worry about that case for now. If we actually end up needing it we can change the format to return an array or something.

**Comment by anderspitman** [comment]
@YangQi007 any progress on this one? No worries if not just checking.

**Comment by YangQi007** [comment]
I committed the change yesterday. Please check it out https://github.com/iobio/iobio-charts/pull/27/commits/be6a5197dbf4de3d103a9bc83bc6a737c0241f33

**Comment by anderspitman** [comment]
@YangQi007 I see the problem. My latest review didn't get submitted properly. I just submitted it. We can pick this up next week.

File: coverage/src/BamViewChart.js [hunk 1]
```diff
@@ -156,6 +172,9 @@ function createBamView(bamHeader, data, element, bamViewControlsElement) {
                 .attr('transform', (d, i) => `translate(${buttons_xScale(d3.sum(bamHeaderArray.slice(0, i), e => e.length)) + margin2.left}, ${margin2.top})`)
                 .on('click', function (event, d) {
                     zoomToChromosome(d.sn);
+                    
+                    // Dispatch custom event from the shadow DOM element
+                    dispatchCustomEvent('chromosomeSelected', { refName: "chr" + d.sn });
```
[the remark this answers is not in the document]
Comment (YangQi007) [reply coverage/src/BamViewChart.js]: I have changed the custom event name. Please check it out again.

**Comment by anderspitman** [comment]
Looks great!

**Pull request merged**